lib/Backend/JITTimeFunctionBody.cpp

```diff
@@ -598,6 +598,32 @@ JITTimeFunctionBody::IsWasmFunction() const
     return m_bodyData.isWasmFunction != FALSE;
 }
 
+bool JITTimeFunctionBody::UsesWAsmJsFastVirtualBuffer() const
+{
+    // Using Fast Virtual Buffer means that bounds checks can be omitted
+#if ENABLE_FAST_ARRAYBUFFER
+#ifdef ENABLE_WASM
+    if (IsWasmFunction())
+    {
+        if (CONFIG_FLAG(WasmFastArray))
```

Comment:
should it be `if(!GetAsmJsInfo()->IsSharedMemory() && CONFIG_FLAG(WasmFastArray))`?

Reply:
Yeah, I've struggled in the past with keeping these settings coherent.
I think I do have a bug, but not the one you think.
By trying to explain it, I realized I messed up. WasmSharedArrayVirtualBuffer does use VirtualAlloc, but not for the 8GB optimization (which is what this function checks), so we should not check this here my bad.
Also, WasmFastArray supersede WasmSharedArrayVirtualBuffer so it's fine to check just that one.